core/metacling/src/TCling.cxx

```diff
@@ -7263,15 +7263,40 @@ const char* TCling::GetIncludePath()
 
    fIncludePath = "";
 
+   std::vector<std::string> filters;
```

Comment:
I will ask @vgvassilev what he thinks as well about these changes

Reply:
This prevents to extract `-I` compiler flags which point to source directory or to some special build subdirectories. One should made these dirs `invisible`,  otherwise they pollute output of `gInterpreter->GetIncludePaths()`

Reply:
Can we instead prevent the filtered dirs from entering into `fInterpreter->GetIncludePaths()`?

Reply:
I did not found exact place where list of includes is produced. 
While there is only two places where includes are used - I just filter them out in these two methods.

Reply:
Another reason why I implement it so - I do not like to put some ROOT special source dirs filtering deep into cling code. In my mind - `TCLing.cxx` is proper place

Reply:
@vgvassilev @Axel-Naumann  Can you please review changes in TCling.cxx and let me know what do you think.

Reply:
> This prevents to extract `-I` compiler flags which point to source directory or to some special build subdirectories. One should made these dirs `invisible`, otherwise they pollute output of `gInterpreter->GetIncludePaths()`

You mean pollute when ROOT is installed but would be accurate when ROOT is built from source and used from within the build directory?

Reply:
I do not know how, but cling able to extract all include paths which were specified during compilation. In my mind, they should not appear in `gInterpreter->GetIncludePaths()` output while source directory may be removed already. Also too long list is not nice - there is roottest which fails when more then 9 includes paths in the list

Reply:
> I did not found exact place where list of includes is produced.

That's not a reason to do it this way. I will help you find it.